src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -474,6 +474,13 @@ protected override IEnumerable<ObjectTypeProperty> BuildProperties()
                         continue;
                     }
                     var prop = CreateProperty(property);
+
+                    // If "Type" property is "String" and "ResourceType" exists in parents properties, skip it since they are the same.
+                    // This only applies for TypeSpec input, for swagger input we have renamed "type" property to "resourceType" in FrameworkTypeUpdater.ValidateAndUpdate
+                    if (property.CSharpName() == "Type" && prop.ValueType.Name == "String" && existingProperties.Contains("ResourceType"))
+                    {
+                        continue;
+                    }
```

Comment:
OK this is really like a workaround.
adding a client.tsp in the resourcemanager to change the client to ResourceType in all trackedresource and resource may be a better idea

Reply:
Ideally, we should rename this for csharp for common-types, but for now we don't have an entry for common-types.
We will need to have an entry to be able to generate `Azure.ResourceManager` from common-types typespec.